api/src/main/java/org/pmiops/workbench/db/dao/UserService.java

```diff
@@ -284,74 +284,74 @@ public DbUser createUser(
   public DbUser createUser(
       String givenName,
       String familyName,
-      String email,
+      String userName,
       String contactEmail,
       String currentPosition,
       String organization,
       String areaOfResearch,
       DbAddress address,
       DbDemographicSurvey demographicSurvey,
       List<DbInstitutionalAffiliation> institutionalAffiliations) {
-    DbUser user = new DbUser();
-    user.setCreationNonce(Math.abs(random.nextLong()));
-    user.setDataAccessLevelEnum(DataAccessLevel.UNREGISTERED);
-    user.setEmail(email);
-    user.setContactEmail(contactEmail);
-    user.setCurrentPosition(currentPosition);
-    user.setOrganization(organization);
-    user.setAreaOfResearch(areaOfResearch);
-    user.setFamilyName(familyName);
-    user.setGivenName(givenName);
-    user.setDisabled(false);
-    user.setAboutYou(null);
-    user.setEmailVerificationStatusEnum(EmailVerificationStatus.UNVERIFIED);
-    user.setAddress(address);
-    user.setDemographicSurvey(demographicSurvey);
+    DbUser dbUser = new DbUser();
+    dbUser.setCreationNonce(Math.abs(random.nextLong()));
+    dbUser.setDataAccessLevelEnum(DataAccessLevel.UNREGISTERED);
+    dbUser.setUsername(userName);
+    dbUser.setContactEmail(contactEmail);
+    dbUser.setCurrentPosition(currentPosition);
+    dbUser.setOrganization(organization);
+    dbUser.setAreaOfResearch(areaOfResearch);
+    dbUser.setFamilyName(familyName);
+    dbUser.setGivenName(givenName);
+    dbUser.setDisabled(false);
+    dbUser.setAboutYou(null);
+    dbUser.setEmailVerificationStatusEnum(EmailVerificationStatus.UNVERIFIED);
+    dbUser.setAddress(address);
+    dbUser.setDemographicSurvey(demographicSurvey);
     // For existing user that do not have address
     if (address != null) {
-      address.setUser(user);
+      address.setUser(dbUser);
     }
-    if (demographicSurvey != null) demographicSurvey.setUser(user);
+    if (demographicSurvey != null) demographicSurvey.setUser(dbUser);
     if (institutionalAffiliations != null) {
-      final DbUser u = user;
+      final DbUser u = dbUser;
```

Comment:
I know this is not part of your change however can we change the variable to a better name than u

Reply:
I think I stole `dbUser` out of this internal scope. I only use single letter variables if they live 2-3 lines or less (and usually only for a lambda).

But that's not the problem here. I don't understand the line `final DbUser u = dbUser;`. I think the original author expected this to allow them to work with a copy of `dbUser`, but objects don't work that way in Java. I'll cut out `u` entierly.

Reply:
The situation is a smell in that we want to capture a variable and later reassign it, which Java doesn't support. So we need a dummy name. I picked a longer one than `u` for now, though for short lambdas I typically am OK with very short variable names.